lib/ansible/modules/cloud/vmware/vmware_guest.py

```diff
@@ -935,22 +986,67 @@ def configure_network(self, vm_obj):
                     if 'gateway' in network_devices[key]:
                         guest_map.adapter.gateway = network_devices[key]['gateway']
 
-                    if self.params.get('domain'):
-                        guest_map.adapter.dnsDomain = self.params['domain']
+                # On Windows, DNS domain is set by network interface
+                if 'domain' in network_devices[key]:
+                    guest_map.adapter.dnsDomain = network_devices[key]['domain']
+                elif self.params.get('domain'):
+                    guest_map.adapter.dnsDomain = self.params['domain']
 
                     adaptermaps.append(guest_map)
 
+        # TODO: Move customizations into separate function
         if vm_obj is None or self.should_deploy_from_template():
             # DNS settings
             globalip = vim.vm.customization.GlobalIPSettings()
             globalip.dnsServerList = self.params['dns_servers']
             globalip.dnsSuffixList = str(self.params['domain'])
 
-            # Hostname settings
-            ident = vim.vm.customization.LinuxPrep()
-            ident.domain = str(self.params['domain'])
-            ident.hostName = vim.vm.customization.FixedName()
-            ident.hostName.name = self.params['name']
+            # If I install a Windows use Sysprep
+            # https://pubs.vmware.com/vi3/sdk/ReferenceGuide/vim.vm.customization.Sysprep.html#field_detail
+            if 'windows' in self.params['guest_id']:
+                ident = vim.vm.customization.Sysprep()
+
+                ident.guiUnattended = vim.vm.customization.GuiUnattended()
+                ident.guiUnattended.autoLogon = self.params['windows_customizations'].get(autologon, False)
+                ident.guiUnattended.autoLogonCount = self.params['windows_customizations'].get(autologoncount, 1)
+
+                ident.identification = vim.vm.customization.Identification()
+
+                ident.userData = vim.vm.customization.UserData()
+                ident.userData.computerName = vim.vm.customization.FixedName()
+                ident.userData.computerName.name = self.params['name']
+                if 'fullname' not in self.params['windows_customizations']:
```

Comment:
I would prefer to move this earlier in the module, so it fails early on. But I also have concerns about type-checking.